cmd/bnc-config/skl.go

```diff
@@ -0,0 +1,154 @@
+	var ifirm meta.FirmwareHistory
+	for _, f := range firmwares {
+		if dr.Model == f.Model && dr.Serial == f.Serial && inWindow(ts, f.Span) {
+			ifirm = f
+			break
+		}
+	}
+
+	if ifirm.Span.Start.IsZero() {
+		return "", fmt.Errorf("no effective firmware version found for this time for %s", mark.Code)
+	}
+
+	antennas := set.InstalledAntennas()
+	var ia meta.InstalledAntenna
+	for _, a := range antennas {
+		if a.Mark == code && inWindow(ts, a.Span) {
+			ia = a
+			break
+		}
+	}
+
+	if ia.Span.Start.IsZero() {
+		return "", fmt.Errorf("no effective installed antenna found for this time for %s", mark.Code)
+	}
+
+	radomes := set.InstalledRadomes()
+	var rad meta.InstalledRadome
+	for _, r := range radomes {
+		if r.Mark == code && inWindow(ts, r.Span) {
+			rad = r
+			break
+		}
+	}
+	// radome is optional, can be nil
+	var radome string
+	if rad.Span.Start.IsZero() {
+		radome = "NONE"
+	} else {
+		radome = rad.Model
+	}
+
+	var monument meta.Monument
+	for _, n := range set.Monuments() {
+		if n.Mark == code && inWindow(ts, n.Span) {
+			monument = n
+			break
+		}
+	}
+
+	var domesNumber string
+	if monument.Span.Start.IsZero() {
+		domesNumber = "UNKNOWN"
+	} else {
+		domesNumber = monument.DomesNumber
+	}
+
+	s := fmt.Sprintf(skeletonFormat,
+		fmt.Sprintf("%s00NZL", code),       // MARKER NAME
+		domesNumber,                        // MARKER NUMBER
+		dr.Serial, dr.Model, ifirm.Version, // REC # / TYPE / VERS
+		ia.Serial, ia.Model, radome, //ANT # / TYPE
+		x, y, z, //APPROX POSITION XYZ
+		ia.Offset.Vertical, ia.Offset.East, ia.Offset.North) // ANTENNA: DELTA H/E/N
+	if mark.Network == "LI" {
+		s += linzComment
+	} else {
+		s += geonetComment
+	}
+
+	return s, nil
+}
+
+const skeletonFormat = `                    OBSERVATION DATA    M (Mixed)           RINEX VERSION / TYPE
+%-60sMARKER NAME
+%-60sMARKER NUMBER
+%-20s%-20s%-20sREC # / TYPE / VERS
+%-20s%-16s%-4s                    ANT # / TYPE
+%-20.4f%-20.4f%-20.4fAPPROX POSITION XYZ
+%-20.4f%-20.4f%-20.4fANTENNA: DELTA H/E/N
+GEODETIC                                                    MARKER TYPE
+GeoNet              GNS                                     OBSERVER / AGENCY
+G   15 C1C C1X C2W C2X C5X L1C L1X L2W L2X L5X S1C S1X S2W  SYS / # / OBS TYPES 
+       S2X S5X                                              SYS / # / OBS TYPES 
+E   15 C1X C5X C6X C7X C8I L1X L5X L6X L7X L8I S1X S5X S6X  SYS / # / OBS TYPES 
+       S7X S8I                                              SYS / # / OBS TYPES
+R   15 C1C C1P C2C C2P C3X L1C L1P L2C L2P L3X S1C S1P S2C  SYS / # / OBS TYPES 
+       S2P S3X                                              SYS / # / OBS TYPES
+C   15 C1X C2I C5X C6I C7I L1X L2I L5X L6I L7I S1X S2I S5X  SYS / # / OBS TYPES 
+       S6I S7I                                              SYS / # / OBS TYPES
+J   18 C1C C1X C1Z C2X C5X C6L L1C L1X L1Z L2X L5X L6L S1C  SYS / # / OBS TYPES 
+       S1X S1Z S2X S5X S6L                                  SYS / # / OBS TYPES
```

Comment:
There seems to be some differences with the trailing spaces on some of these template lines. I assume they either should all line up, or all be trimmed.